runtime/domain.c

```diff
@@ -235,7 +232,7 @@ static dom_internal* next_free_domain() {
 
 #ifdef __APPLE__
 /* OSX has issues with dynamic loading + exported TLS.
-    This is slower but works */
```

Comment:
Unrelated to this PR, but: it would be good to check if this macOS workaround is still needed.  Reading up on [a recent TLS implementation survey](https://maskray.me/blog/2021-02-14-all-about-thread-local-storage) seems to indicate that the situation has improved there (for `thread_local` anyway).

Reply:
Indeed, should we open an issue to keep track of this?